### PR TITLE
Create new diff component (AI-2737)

### PR DESCRIPTION
**Pull request by annloouise**

### CODE REVIEW

File: src/components/diff/ai-diff.js [hunk 1]
```diff
@@ -27,77 +37,130 @@ customElements.define('ai-diff', class AiDiff extends AIHTMLElement {
...
+    this.htmlDiff.innerHTML = Diff2Html.getPrettyHtml(diff, {
+      inputFormat: 'diff', showFiles: false, matching: 'lines', outputFormat: view,
+    });
+
+    if (view === 'side-by-side') {
+      const firstScrollArea = this.htmlDiff.querySelector('.d2h-files-diff').children[0];
+      const secondScrollArea = this.htmlDiff.querySelector('.d2h-files-diff').children[1];
+      const self = this;
+
+      this.firstTable = firstScrollArea.querySelector('.d2h-diff-table');
```
Comment (bjornwiberg) [inline src/components/diff/ai-diff.js]: why `this` here when you assigned it to `self` on row `73`?

then you use is as `self` on row `79-82`?

File: src/components/diff/ai-diff.js [hunk 2]
```diff
@@ -27,77 +37,130 @@ customElements.define('ai-diff', class AiDiff extends AIHTMLElement {
...
+      this.secondTable = secondScrollArea.querySelector('.d2h-diff-table');
+      self.adjustScrollbars();
+
+      window.onresize = function resizeTables() {
+        self.firstTable.style.width = '100%';
+        self.secondTable.style.width = '100%';
+        self.adjustScrollbars();
+      };
+
+      firstScrollArea.onscroll = function syncScroll() {
```
Comment (bjornwiberg) [inline src/components/diff/ai-diff.js]:
```suggestion
       firstScrollArea.onscroll = () => { secondScrollArea.scrollLeft = firstScrollArea.scrollLeft; };
       secondScrollArea.onscroll = () => { firstScrollArea.scrollLeft = secondScrollArea.scrollLeft; };
```

File: src/components/diff/ai-diff.js [hunk 3]
```diff
@@ -27,77 +37,130 @@ customElements.define('ai-diff', class AiDiff extends AIHTMLElement {
...
+  * @param {String} diff Unified diff patch
+  * @param {String} view Type of view, can be 'line-by-line' or 'side-by-side'
+  */
+  createHtmlDiff(diff, view) {
+    this.htmlDiff.innerHTML = Diff2Html.getPrettyHtml(diff, {
+      inputFormat: 'diff', showFiles: false, matching: 'lines', outputFormat: view,
+    });
+
+    if (view === 'side-by-side') {
+      const firstScrollArea = this.htmlDiff.querySelector('.d2h-files-diff').children[0];
```
Comment (bjornwiberg) [inline src/components/diff/ai-diff.js]:
```suggestion
      const [firstScrollArea, secondScrollArea] = this.htmlDiff.querySelector('.d2h-files-diff').children;
```

File: src/components/diff/ai-diff.js [hunk 4]
```diff
@@ -27,77 +37,130 @@ customElements.define('ai-diff', class AiDiff extends AIHTMLElement {
...
+  }
 
-    return '';
+  set targetobject(val) {
+    this.handleSetAttribute('targetobject', val);
   }
 
   template() {
     return html`
+      <style>${diffCss}</style>
```
Comment (aihazm) [inline src/components/diff/ai-diff.js]: we don't need multiple style tags here, the poly-fill will not be happy about that (especially IE/EDGE)

File: src/components/diff/ai-diff.js (hunk 1 above)
Comment (aihazm) [reply src/components/diff/ai-diff.js]: alternatively, use only `self` in the onresize callback

Comment (bjornwiberg) [reply src/components/diff/ai-diff.js]: or just
```javascript
      window.onresize = () => {
        this.firstTable.style.width = '100%';
        this.secondTable.style.width = '100%';
        this.adjustScrollbars();
      };
```

**Pull request merged**